python-main.js

```diff
@@ -1075,31 +1075,56 @@ function web_editor(config) {
             $("#command-disconnect > .roundlabel").text(config["translate"]["static-strings"]["buttons"]["command-disconnect"]["label"]);
             $("#command-disconnect").attr("title", config["translate"]["static-strings"]["buttons"]["command-disconnect"]["title"]);
 
-            // Change download to flash
-            $("#command-download").attr("id", "command-flash");
-            $("#command-flash > .roundlabel").text(config["translate"]["static-strings"]["buttons"]["command-flash"]["label"]);
-            $("#command-flash").attr("title", config["translate"]["static-strings"]["buttons"]["command-flash"]["title"]);
-
             if (serial){
-              doSerial();
+                doSerial();
             }
         })
-        .catch(function(err) {
-            console.log("Error connecting: " + err);
-            $("#flashing-overlay-container").css("display", "flex");
-            $("#flashing-info").addClass('hidden');
-
-            // If micro:bit does not support dapjs
-            if (err.message === "No valid interfaces found."){
-                $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["update-req"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            } else if (err.message === "Unable to claim interface.") {
-                $("#flashing-overlay-error").html('<div>'+ config["translate"]["webusb"]["err"]["clear-connect"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            }
-
-            $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["restart-microbit"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
+        .catch(errorHandler);
+    
+        // WebUSB Disconnect Events
+        navigator.usb.addEventListener('disconnect', event => {
+            var error = {"message": config["translate"]["webusb"]["err"]["device-disconnected"]};
+            errorHandler(error);
         });
+
+    }
+
+    function errorHandler(err) {
```

Comment:
Since this function is directly inside `web_editor()` (instead of inside a `doConnect`/`doFlash` or similar) maybe it needs to be more explicit about what kind of error handler it is, should we rename this to `webusbErrorHandler`?

Reply:
Changed in c61cad7bad572e535882845f5cea61ea1293c6e5.